raiden/transfer/views.py

```diff
@@ -517,7 +517,10 @@ def get_transfer_role(chain_state: ChainState, node_address: AddressHex, secreth
     in a transfer. If a transfer task is not found for the secrethash then the
     function returns None
     """
-    task = chain_state.payment_mapping[node_address].secrethashes_to_task.get(secrethash)
+    task = None
```

Comment:
Some comments here:

1. why this was needed? 
2. its ok to protect ourselves of an exception but this scenario cant happen, if happens, we have a major issue, so i prefeer to keep the code as it was
3. i think that if a PR is for solving tests and we modify something of the core of lumino, that must be another pr (with, probably, another round of manual and unit tests)

Reply:
This was because you have chain_state.payment_mapping as a dict empty sometimes and accessing a dict without that key blows the code, the background of why this dict is empty some times i don't have it, but seems to me that when you delete the task you use del and del deletes the key also so you loose the access to the object with the secret hashes too. The tests were failing before this so that's why i added that.

Reply:
hmm but that situation of having the empty dict and accesing it, is related with some test? 

and if the answer is yes, why wasnt failing previously?

Reply:
Because before that you didn't access the dict with the node address, is like a nesting problem, before this you had something like this:
```python
chain_state.secrethashes_to_task.get(secrethash)
```
where secrethash could be or not in the dict and that was ok, since it never will fail, because if the hash wasn't there the if will check it, but now you have:
```python
chain_state.payment_mapping[node_address].secrethashes_to_task.get(secrethash)
```
where chain_state.payment_mapping can be = {} so the first "get" chain_state.payment_mapping[node_address] will be None and None can't be accessed directly so the rest of the call .secrethashes_to_task.get(secrethash) will fail for those cases.

Reply:
Maybe a better solution to this problem could be to make sure that when you create the dict chain_state.payment_mapping you add the addresses to the dict there, but i think that you don't have the addresses there so you need to make this check here.